spyder/app/mainwindow.py

```diff
@@ -1141,8 +1141,18 @@ def post_visible_setup(self):
         Actions to be performed only after the main window's `show` method
         is triggered.
         """
-        # Process pending events and hide splash before loading the
-        # previous session.
+        # Populate `Panes > View` menu.
+        # This **MUST** be done before restoring the last visible plugins, so
+        # that works as expected.
+        self.layouts.create_plugins_menu()
+
+        # Restore last visible plugins.
+        # This **MUST** be done before running on_mainwindow_visible for
+        # plugins so that the user doesn't experience sudden jumps in the
+        # interface.
+        self.layouts.restore_visible_plugins()
```

Comment:
Perhaps we could move this to `layouts.on_mainwindow_visible` in order to decrease the possibility of it being changed in the future. What do you think @dalthviz?

Reply:
And then here we'd only call `self.layouts.on_mainwindow_visible` before calling `on_mainwindow_visible` for the rest of the plugins.

Reply:
Sure :+1: